stages/org.osbuild.oscap.remediation

```diff
@@ -89,34 +86,40 @@ SCHEMA = """
 }
 """
 
-# Map containing the translations between
-# our configuration names and the OpenScap
-# configuration names.
-OSCAP_OPTIONS = {
-    "datastream" : "OSCAP_REMEDIATE_DS",
-    "profile_id" : "OSCAP_REMEDIATE_PROFILE_ID",
-    "datastream_id" : "OSCAP_REMEDIATE_DATASTREAM_ID",
-    "xccdf_id" : "OSCAP_REMEDIATE_XCCDF_ID",
-    "benchmark_id" : "OSCAP_REMEDIATE_BENCHMARK_ID",
-    "tailoring" : "OSCAP_REMEDIATE_TAILORING",
-    "tailoring_id" : "OSCAP_REMEDIATE_TAILORING_ID",
-    "arf_result" : "OSCAP_REMEDIATE_ARF_RESULT",
-    "html_report" : "OSCAP_REMEDIATE_HTML_REPORT",
-    "verbose_log" : "OSCAP_REMEDIATE_VERBOSE_LOG",
-    "verbose_level" : "OSCAP_REMEDIATE_VERBOSE_LEVEL",
-}
+XML_RESULTS = "eval_remediate_results.xml"
+HTML_REPORT = "eval_remediate_report.html"
 
 def main(tree, options):
-    filename = options["filename"]
     config = options["config"]
-
-    with open(f"{tree}/{filename}", "w", encoding="utf-8") as f:
-        for ours, theirs in OSCAP_OPTIONS.items():
-            value = config.get(ours)
-            if value:
-                f.write(f"{theirs}={value}\n")
-
-    os.symlink(f"{filename}", f"{tree}/system-update")
+    datastream = config.get("datastream")
+    profile = config.get("profile_id")
+    ds_id = config.get("datastream_id")
+    xccdf_id = config.get("xccdf_id")
+    xml_results = config.get("xml_results", XML_RESULTS)
+    html_report = config.get("html_report", HTML_REPORT)
+
+    os.mkdir(f"{tree}/openscap_data")
+
+    # run openscap in chroot on the image tree
+    cmd = [
+      "/usr/sbin/chroot", tree,
+      "/usr/bin/oscap", "xccdf", "eval",
+      "--remediate", "--profile", profile,
+      "--results", f"/openscap_data/{xml_results}",
+      "--report", f"/openscap_data/{html_report}"
+    ]
+
+    if not ds_id is None:
+        cmd.extend(["--datastream-id", ds_id])
+
+    if not xccdf_id is None:
+        cmd.extend(["--xccdf-id", xccdf_id])
+
+    cmd.append(datastream)
+
+    # check false since oscap has return code of 2
+    # if any oscap rule has failed or is unknown
+    subprocess.run(cmd, encoding="utf-8", stdout=sys.stderr, check=False)
```

Comment:
I think I would then check the return value and print something for `2` and fail for `1`